src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -1150,6 +1150,12 @@ internal HttpListenerContext HandleAuthentication(RequestContextBase memoryBlob,
                             NetEventSource.Info(this, $"Got principal: {principal}, IdentityName: {principal.Identity.Name} for creating request.");
                         }
 
+                        // Prefix SPNEGO token with Negotiate scheme per RFC 4559
+                        if (!String.IsNullOrEmpty(outBlob) && headerScheme == AuthenticationSchemes.Negotiate)
```

Comment:
Does outblob ever have a value at this stage for protocols besides Negotiate? It doesn't look like it. Move this up into the negotiate code block. See:
https://github.com/mattpwhite/corefx/blob/3ef8e6cd59787df1859e62bd69518938db8127c8/src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs#L1011
https://github.com/mattpwhite/corefx/blob/3ef8e6cd59787df1859e62bd69518938db8127c8/src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs#L1103-L1107

Reply:
Maybe, I'll have a look.  This method is umm... sprawling.  It has been a little bit since I actually wrote this, so I'll need a bit to re-read everything.

Reply:
I think we only end up in that branch at 1100 if we haven't completed establishing the context (meaning we're going to 401 and send an NTLM challenge or we need to exchange another round of SPNEGO blobs for negotiate).  In the mutual auth case, `context.IsCompleted` should be true (it is from the server's perspective) we go down the other path at 1018.  I guess it could go right there, is that what you're suggesting?

Reply:
Yes, right around 1011.